module/modmanager/manager.go

```diff
@@ -211,6 +211,21 @@ func (mgr *Manager) Close(ctx context.Context) error {
 	return err
 }
 
+// Kill kills module processes. This is best effort as we do not
```

Comment:
> Kill kills modules processes.

Per my nits on the other PR, we're technically killing module process groups here. Could we make sure that's at least clear in documentation here and wherever else we're calling `managedProcess.Kill` and describing what we're doing? The group vs. process thing led to some initial (and perhaps continued) confusion for me, so, while this looks pretty good code-wise, I'd love to not be confused looking back at this stuff.

Reply:
updated comments